src/domain/opendatafabric/src/identity/dataset_refs.rs

```diff
@@ -740,3 +741,126 @@ impl std::cmp::PartialOrd for DatasetRefAny {
         Some(self.cmp(other))
     }
 }
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum DatasetRefPattern {
+    Ref(DatasetRef),
+    Pattern(Option<AccountName>, DatasetNamePattern),
+}
+
+impl DatasetRefPattern {
+    pub fn match_pattern(dataset_ref: &str, pattern: &str) -> Result<bool, InvalidPatternError> {
+        Like::<false>::like(dataset_ref, pattern)
```

Comment:
@zaychenko-sergei 
This actually brings up a very important point - should comparing all aliases be case-insensitive?

Since we already use dataset names as subdomains sometimes - I think the answer is YES, and we should amend both the spec and equality comparisons of names/aliases.

Reply:
Created a proposal https://github.com/kamu-data/kamu-cli/issues/469